lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1280,7 +1290,11 @@ def _post_start_init(self, event: _StartOpenSearch):  # noqa: C901
                 event.defer()
                 return
 
-        self.peers_data.put(Scope.UNIT, "started", True)
+        # Add a timestamp instead of just a boolean to be able to trigger
+        # a relation changed event in case of restarts
+        # if you remove the flag and add it again in the same hook the
+        # relation changed event won't be triggered
```

Comment:
I think no need to add this, as the rel data change only happens after the hook completes as per juju functioning